src/ansys/sound/core/sound_composer/track.py

```diff
@@ -0,0 +1,337 @@
+            str_source = "Not set"
+
+        return (
+            f"{self.name}\n"
+            f"\tSource: {str_source}\n"
+            f"\tFilter: {"Set" if self.filter is not None else "Not set"}"
+        )
+
+    @property
+    def name(self) -> str:
+        """Name of the track."""
+        return self.__name
+
+    @name.setter
+    def name(self, string: str):
+        """Set the track name."""
+        self.__name = string
+
+    @property
+    def gain(self) -> float:
+        """Track gain in dB.
+
+        Gain in dB to apply to the synthesized signal of the track.
+        """
+        return self.__gain
+
+    @gain.setter
+    def gain(self, value: float):
+        """Set the track gain."""
+        self.__gain = value
+
+    @property
+    def source(self) -> AnySourceType:
+        """Source object associated with the track.
+
+        The source of the track is used to synthesize the corresponding signal. Its type can be
+        either :class:`SourceSpectrum`, :class:`SourceBroadbandNoise`,
+        :class:`SourceBroadbandNoiseTwoParameters`, :class:`SourceHarmonics`,
+        :class:`SourceHarmoncisTwoParameters`, or :class:`SourceAudio`.
+        """
+        return self.__source
+
+    @source.setter
+    def source(self, obj: AnySourceType):
+        """Set the track source."""
+        if (obj is not None) and (not (isinstance(obj, AnySourceType))):
+            raise PyAnsysSoundException(
+                "Specified source must have a valid type (SourceSpectrum, SourceBroadbandNoise, "
+                "SourceBroadbandNoiseTwoParameters, SourceHarmonics, "
+                "SourceHarmoncisTwoParameters, or SourceAudio)."
+            )
+        self.__source = obj
+
+    @property
+    def filter(self) -> Filter:
+        """Filter object of the track."""
+        return self.__filter
+
+    @filter.setter
+    def filter(self, obj: Filter):
+        """Set the track filter."""
+        if (obj is not None) and (not (isinstance(obj, Filter))):
+            raise PyAnsysSoundException("Specified filter must be of type Filter.")
+        self.__filter = obj
+
+    def process(self, sampling_frequency: float = 44100.0):
+        """Generate the signal of the track, using the current source and filter.
+
+        Parameters
+        ----------
+        sampling_frequency : float, default: 44100.0
+            Sampling frequency of the generated sound in Hz.
+        """
+        if sampling_frequency <= 0.0:
+            raise PyAnsysSoundException("Sampling frequency must be strictly positive.")
+
+        if self.source is None:
+            raise PyAnsysSoundException(f"Source is not set. Use {__class__.__name__}.source.")
+
+        self.source.process(sampling_frequency)
+        signal = self.source.get_output()
+
+        if self.filter is not None:
+            self.filter.signal = signal
+            self.filter.process()
+            signal = self.filter.get_output()
+
+        if self.gain != 0.0:
```

Comment:
@ansaminard it's minor here but testing strict equality `==` or `!=` on float numbers is usually not a good practice

But you can leave it that way

Reply:
I could have even skipped the test altogether, but I felt it made sense to not call another operator if it is unnecessary